src/main/java/com/cloudbees/jenkins/plugins/sshagent/mina/MinaRemoteAgentStarter.java

```diff
@@ -24,15 +24,23 @@
 
 package com.cloudbees.jenkins.plugins.sshagent.mina;
 
+import jenkins.security.MasterToSlaveCallable;
+
 import com.cloudbees.jenkins.plugins.sshagent.RemoteAgent;
+
 import hudson.model.TaskListener;
-import hudson.remoting.Callable;
 import hudson.remoting.Channel;
 
 /**
  * Callable to start the remote agent.
  */
-public class MinaRemoteAgentStarter implements Callable<RemoteAgent, Throwable> {
+public class MinaRemoteAgentStarter extends MasterToSlaveCallable<RemoteAgent, Throwable> {
+
+    /**
+     * 
+     */
+    private static final long serialVersionUID = 2909967505958103050L;
```

Comment:
Well then that's a :bug: as the UID should be the UID from before your modifications. If it is the UID from after your modifications then that can cause issues